NitroxServer/GameLogic/Spawning/EntitySpawner.cs

```diff
@@ -106,5 +110,74 @@ private void SpawnEntities()
                 }
             }
         }
+
+        private bool GetDataFiles(out string lootDistributions, out Dictionary<string, WorldEntityInfo> worldEntityData)
+        {
+            lootDistributions = "";
+            worldEntityData = new Dictionary<string, WorldEntityInfo>();
+            string resourcesPath = "";
+            string gameResourcesPath = Path.Combine(SteamFinder.FindSteamGamePath(264710, "Subnautica"), "Subnautica_Data/resources.assets");
+            AssetsFile resourcesFile;
+            if (File.Exists(gameResourcesPath))
+            {
+                resourcesPath = gameResourcesPath;
+            }
+            else if (File.Exists("../resources.assets"))
+            {
+                resourcesPath = Path.GetFullPath("../resources.assets");
+            }
+            else if (File.Exists("resources.assets"))
+            {
+                resourcesPath = Path.GetFullPath("resources.assets");
+            }
+            else
+            {
+                Log.Error("Could not find resources.assets in parent or current directory.", new FileNotFoundException());
+                return false;
+            }
+
+            using (FileStream resStream = new FileStream(resourcesPath, FileMode.Open))
+            {
+                resourcesFile = new AssetsFile(new AssetsFileReader(resStream));
+                AssetsFileTable resourcesFileTable = new AssetsFileTable(resourcesFile);
+                foreach (AssetFileInfoEx afi in resourcesFileTable.pAssetFileInfo)
+                {
+                    if (afi.curFileType == 0x31) //TextAsset
```

Comment:
I can compile up a classdatabase to use so that all the serialization data comes from file rather than manually coding the names and the readxxx methods. It's another file though, and requires a little more setup.

Reply:
As you might know, I'm not a fan of precompiled files, especially if it's not integrated with the VS project and generated/updated when the project is rebuilt. That way we rely on a file that some contributor to the project once made and added. Of course, providing the tools and documentation helps, but it's still an extra dependency we have to watch out for.

Anyway, that wasn't really the point. If you can migrate these values to an enum, I'm happy.
```csharp
enum AssetFileType : int {
    TextAsset = 0x31,
    MonoBehaviour = 0x72
}
```
Besides, are these values fixed for all (Unity) asset files, or is this specific to Subnautica?

Finally, as you know this has sparked some discussion in Discord regarding rolling our own code that effectively does the same as the game. I still want to see if we can look further into reusing the game and library code, as this saves us a lot of time and dependency maintenance.

Reply:
I understood the whole enum thing, I probably was going off the subject. I usually only use things like enums when they are used multilple times and having an enum just there and only being used once seems kind of weird to me. The values are halfway fixed. The TextAsset will always read that way and the MonoBehaviour as you know can change if they decide to change any fields in that class. Since they are using a custom build of unity, it is very very unlikely that they will use a different unity version, and the IDs will remain the same.
The code that reads these two values comes from the assets files, and these are read natively by the player exe, not by the Assembly-CSharp, otherwise I would've just used the method in unity.

Reply:
I agree, but I also prefer to use them to hide such magic values, even if they are only used once. Makes the intent clear without having to specify it in a comment, or looking up what value you need elsewhere.

Thanks for the info on Unity, I think it's okay to accept the ID's for now, assuming they won't change anytime soon.

Reply:
The general rule of thumb is to use an enum whenever you need to define a static range of values that a specific value *must* be within. They also have the added benefit of being (mostly) human readable. However; it is an implementation detail that clarifies a specific intent.

The catch here is that we cannot enforce what the value on the provided instance *should* be. We can only write some control statements to react to possible values that it *could* be. As such, I do not think that an enum would be the best choice in this context.

There still remains the issue of a hard coded value within the control statement itself - which is generally considered a bast practice for readability concerns. To address that, consider creating some class-level constants that describes the value that you are storing within it and reference those in your code instead. 

Ex:
```c#
class Demo {
private const int TEXT_ASSET_TYPE = 0x31;

   public void doThis(instance) {
      if(instance.type == TEXT_ASSET_TYPE)...
   }
}
```

Reply:
I think I agree with this more. It's only two lines and I don't have to cast the enum to a uint

Reply:
Indeed, `const` is the alternative I was looking for, since there is currently no function or variable that can only accept that enum type to specify exactly what range *or set* of values is allowed. `Const` does exactly what I was looking for in the first comment: a human-readable 'accessor' to a magic number.

Reply:
Bingo! Using an enum in this case would be like using a hammer as a paper weight. No one can argue that itserves the function perfectly, but everyone else that walks into the room is really confused about why there is a hammer sitting on a stack of papers on your desk...